### ISSUE TITLE
personal project.toml question

### ISSUE THREAD

**Issue by nathanrboyer** [MakieOrg/Makie.jl#2332]
When writing my own package with Makie as a dependency, I am confused about what to include in my Project.toml regarding the main package vs. the backends.

I `add` and `using` a backend, so that is what appears in the `[deps]` section. However, I don't know how to keep track of the backend version or breaking changes to it since the Github repository tracks changes to Makie itself. Writing, for example, `GLMakie = 0.18` under `[compat]` doesn't work, since the GLMakie version is different from the Makie version.
![image](https://user-images.githubusercontent.com/65452054/195627678-f14302a5-4631-4b6f-829b-b1115a2fb6f7.png)

What is the suggested way to make Makie a dependency of a package?

Can I somehow make Makie (main package) a dependency of my package instead of a specific backend?
1. so that I can reference Github when checking compatibility
2. so that I can change which backend my package uses without changing my Project.toml

Can you add this topic to the documentation? I was not able to find the answer there.

**Comment by Moelf** [comment]
include what you use, since you can't use Makie without a backend, it's pointless to only include Makie but not backend.

**Comment by nathanrboyer** [comment]
In my case, I have a `main` function that runs a whole simulation and calls a `plot_results` function at the end, so the users should not be too involved in the plotting details. I am more wondering what I should do as a package maintainer to quickly update and check version numbers when Makie and/or its backends update, and how that should be written into my package.

**Comment by Moelf** [comment]
If you plot, you need to pick a backend, that's what I imagined happening.

If you're providing plottable recipes, you only need Makie and user can pick backend.

Another possibility is to just use Requires.jl

**Comment by nathanrboyer** [comment]
Okay, once I pick a Makie backend, how does the backend version relate to the Makie version? Can I filter the release notes by backend?

**Comment by Moelf** [comment]
why do you care? you treat the backend as a package, the Makie/Makie core dependency will sort out itself

**Comment by SimonDanisch** [comment]
>Okay, once I pick a Makie backend, how does the backend version relate to the Makie version?

Yeah, so if you really need to pick a backend and don't want users interacting with it, you should just depend on the backend.
They all re-export all of Makie, so you don't really need a using Makie, or a direct dependency on Makie.

You get the version by looking at the backends Project.toml of the Makie version you're interested in:
https://github.com/MakieOrg/Makie.jl/blob/v0.18.0/GLMakie/Project.toml

**Comment by nathanrboyer** [comment]
Okay, it is just confusing that I have to look through the code to translate the version number from Github Releases and Discourse Package Announcements for Makie to the associated backend version. Why not just synchronize all the version numbers?

**Comment by jkrumbiegel** [comment]
Because we sometimes release backend upgrades separately.
